packages/api-client/src/odsql/index.ts

```diff
@@ -164,7 +164,8 @@ export const string = (value: string) => JSON.stringify(value);
 
 export const dateTime = (date: Date) => `date'${date.toISOString()}'`;
 
-export const date = (date: Date) => `date'${date.toISOString().split('T')[0]}'`;
+export const date = (year: number, month?: number, day?: number) =>
+    `date'${year}${month !== undefined ? `-${month}` : ''}${day !== undefined ? `-${day}` : ''}'`;
```

Comment:
Apparently the API V2 considers valid dates to be similar to ISO, so `2021-01` and not `2021-1` (https://public.opendatasoft.com/api/v2/catalog/datasets/?where=modified%3Edate%272021-1-1%27). 

We'd have to leftpad months and days. I heard there's a good package in NPM for that 😈. But since we're not supporting IE, `padStart` will work for us. Something like `String(day).padStart(2, "0")`.

Also, i'm thinking of situation where you already have the date as an ISO string (which should be frequent IMO, as most UI components can provide you with that). Maybe we can also allow taking a string directly and use it as-is? (it won't be as "structured", but it will still solve the issue of phantom timezones).
Otherwise, it's a bit tricky currently, as you'll have to do something like `date(...myIsoDate.split('-').map(Number))`

Reply:
If you have a date as a ISO string, you can directly insert it in the where clause normally.